core/src/main/java/de/danoeh/antennapod/core/service/download/NewEpisodesNotification.java

```diff
@@ -0,0 +1,94 @@
+package de.danoeh.antennapod.core.service.download;
+
+import android.app.Notification;
+import android.app.PendingIntent;
+import android.content.ComponentName;
+import android.content.Context;
+import android.content.Intent;
+import android.content.res.Resources;
+
+import androidx.core.app.NotificationCompat;
+import androidx.core.app.NotificationManagerCompat;
+
+import de.danoeh.antennapod.core.R;
+import de.danoeh.antennapod.core.feed.Feed;
+import de.danoeh.antennapod.core.feed.FeedPreferences;
+import de.danoeh.antennapod.core.preferences.UserPreferences;
+import de.danoeh.antennapod.core.storage.DBReader;
+import de.danoeh.antennapod.core.storage.PodDBAdapter;
+import de.danoeh.antennapod.core.util.gui.NotificationUtils;
+
+public class NewEpisodesNotification {
+    static final String GROUP_KEY = "de.danoeh.antennapod.EPISODES";
+
+    private final int lastEpisodeCount;
+    private final boolean shouldShowNotification;
+
+    public NewEpisodesNotification(Long feedId) {
+        Feed feed = DBReader.getFeed(feedId);
+
+        FeedPreferences prefs = feed.getPreferences();
+        if (!prefs.getKeepUpdated() || !prefs.getShowEpisodeNotification()) {
+            shouldShowNotification = false;
+            lastEpisodeCount = -1;
+            return;
+        }
+
+        lastEpisodeCount = getNewEpisodeCount(feedId);
+        shouldShowNotification = true;
+    }
+
+    public void showIfNeeded(Context context, Feed feed) {
+        if (!shouldShowNotification) {
+            return;
+        }
+
+        long feedId = feed.getId();
+        int newEpisodes = getNewEpisodeCount(feedId);
+
+        if (newEpisodes > lastEpisodeCount) {
+            NotificationManagerCompat notificationManager = NotificationManagerCompat.from(context);
+            showNotification(newEpisodes, feed, context, notificationManager);
+        }
+    }
+
+    private static void showNotification(int newEpisodes, Feed feed, Context context,
+                                         NotificationManagerCompat notificationManager) {
+        Resources res = context.getResources();
+        String text = res.getQuantityString(
+                R.plurals.new_episode_notification_message, newEpisodes, newEpisodes, feed.getTitle()
+        );
+        String title = res.getQuantityString(R.plurals.new_episode_notification_title, newEpisodes);
+
+        Intent intent = new Intent();
+        intent.setComponent(new ComponentName(context, "de.danoeh.antennapod.activity.MainActivity"));
+
+        intent.setFlags(Intent.FLAG_ACTIVITY_NEW_TASK | Intent.FLAG_ACTIVITY_CLEAR_TASK);
+        intent.putExtra("fragment_feed_id", feed.getId());
+        PendingIntent pendingIntent = PendingIntent.getActivity(context, 0, intent, 0);
+
+        Notification notification = new NotificationCompat.Builder(
+                context, NotificationUtils.CHANNEL_ID_EPISODE_NOTIFICATIONS)
+                .setSmallIcon(R.drawable.ic_notification)
```

Comment:
Just created some mock-ups for the notification icons. Do you have another idea? If not, I can draw the images and add them to the branding repo.

Player:
<img width="400" src="https://user-images.githubusercontent.com/5811634/100925006-b2623a80-34e1-11eb-8414-996a66cb2239.png" />

New episodes: (this PR)
<img width="400" src="https://user-images.githubusercontent.com/5811634/100925003-b1310d80-34e1-11eb-80a2-75e97eb857e6.png" />

Auto download completed: (another PR)
<img width="400" src="https://user-images.githubusercontent.com/5811634/100924997-afffe080-34e1-11eb-8cac-295c99578847.png" />